src/common/common.ts

```diff
@@ -396,43 +445,230 @@ export async function getVerifiableCredentialApiConfiguration(
+    },
+    get size() {
+      return store.size;
+    },
+    // For backwards compatibility the dataset properties
+    // SHOULD NOT be included when we JSON.stringify the object
+    toJSON() {
+      return vc;
+    },
+  });
+}
+
+/**
+ * @hidden
+ */
+export async function verifiableCredentialToDataset<T extends { id?: string }>(
+  vc: T,
+  options?: ParseOptions & {
+    includeVcProperties: true;
+  },
+): Promise<T & DatasetWithId>;
+export async function verifiableCredentialToDataset<T extends { id?: string }>(
+  vc: T,
+  options?: ParseOptions & {
+    includeVcProperties?: boolean;
+  },
+): Promise<DatasetWithId>;
+export async function verifiableCredentialToDataset<T extends { id?: string }>(
+  vc: T,
+  options?: ParseOptions & {
+    includeVcProperties?: boolean;
+  },
+): Promise<DatasetWithId> {
+  let store: DatasetCore;
+  try {
+    store = await jsonLdToStore(vc, options);
+  } catch (e) {
+    throw new Error(
+      `Parsing the Verifiable Credential as JSON-LD failed: ${e}`,
+    );
+  }
+
+  if (typeof vc.id !== "string") {
+    throw new Error(
+      `Expected vc.id to be a string, found [${
+        vc.id
+      }] of type [${typeof vc.id}] on ${JSON.stringify(vc, null, 2)}`,
+    );
+  }
+
+  return internal_applyDataset(vc as { id: string }, store, options);
+}
+
+// eslint-disable-next-line camelcase
+export async function internal_getVerifiableCredentialFromResponse(
+  vcUrl: UrlString | undefined,
+  response: Response,
+  options?: ParseOptions & {
+    returnLegacyJsonld?: true;
+    normalize?: (object: VerifiableCredentialBase) => VerifiableCredentialBase;
+  },
+): Promise<VerifiableCredential>;
+export async function internal_getVerifiableCredentialFromResponse(
+  vcUrl: UrlString | undefined,
+  response: Response,
+  options?: ParseOptions & {
+    returnLegacyJsonld?: boolean;
+  },
+): Promise<DatasetWithId>;
+export async function internal_getVerifiableCredentialFromResponse(
+  vcUrlInput: UrlString | undefined,
+  response: Response,
+  options?: ParseOptions & {
+    returnLegacyJsonld?: boolean;
+    normalize?: (object: VerifiableCredentialBase) => VerifiableCredentialBase;
+  },
+): Promise<DatasetWithId> {
+  const returnLegacy = options?.returnLegacyJsonld !== false;
+  let vc: unknown | VerifiableCredentialBase;
+  let vcUrl = vcUrlInput;
+  try {
+    vc = await response.json();
+
+    if (typeof vcUrl !== "string") {
+      if (!isUnknownObject(vc) || !("id" in vc) || typeof vc.id !== "string") {
+        throw new Error("Cannot establish id of verifiable credential");
+      }
+      vcUrl = vc.id;
+    }
+
+    if (returnLegacy) {
+      vc = normalizeVc(vc);
+    }
+  } catch (e) {
+    throw new Error(
+      `Parsing the Verifiable Credential [${vcUrl}] as JSON failed: ${e}`,
+    );
+  }
+
+  if (returnLegacy) {
+    if (!isVerifiableCredential(vc)) {
+      throw new Error(
+        `The value received from [${vcUrl}] is not a Verifiable Credential`,
+      );
+    }
```

Comment:
Note that this change will result in errors like `cannot read property proof of undefined` rather than `Parsing the Verifiable Credential [http://some.vc/] as JSON failed: ${cannot read property proof of undefined}`.

If you are ok with this dropping of context in the error message then we can go ahead with this change.

Reply:
You're right, the former is clearer. Let's keep things as they are, but maybe add a comment to explain, because I'll forget about this and want to have the same change in 6 months time ^^